### PR TITLE
3.0: Extract formatTreeList() from findTreeList()

### PR DESCRIPTION
**Pull request by dereuromark**
3.x port of https://github.com/cakephp/cakephp/pull/6704

### CODE REVIEW

File: src/ORM/Behavior/TreeBehavior.php [hunk 1]
```diff
@@ -441,35 +442,30 @@ function ($field) {
...
      *
-     * - keyPath: A dot separated path to fetch the field to use for the array key, or a closure to
-     *  return the key out of the provided row.
-     * - valuePath: A dot separated path to fetch the field to use for the array value, or a closure to
-     *  return the value out of the provided row.
-     * - spacer: A string to be used as prefix for denoting the depth in the tree for each item
+     * - keyPath: A dot separated path to the field that will be the result array key, or a closure to
+     *   return the key from the provided row.
+     * - valuePath: A dot separated path to the field that is the array's value, or a closure to
+     *   return the value from the provided row.
```
Comment (markstory) [inline src/ORM/Behavior/TreeBehavior.php]: Is there any test coverage for keyPath and valuePath as a callable?

Comment (lorenzo) [inline src/ORM/Behavior/TreeBehavior.php]: yes, there is

File: src/ORM/Behavior/TreeBehavior.php [hunk 2]
```diff
@@ -441,35 +442,30 @@ function ($field) {
...
      *
-     * @param \Cake\ORM\Query $query Query.
-     * @param array $options Array of options as described above
-     * @return \Cake\ORM\Query
+     * @param \Cake\ORM\Query $query The query object to format.
+     * @param array $options Array of options as described above.
+     * @return \Cake\ORM\Query Augmented query.
      */
-    public function findTreeList(Query $query, array $options)
+    public function formatTreeList(Query $query, array $options = [])
```
Comment (lorenzo) [inline src/ORM/Behavior/TreeBehavior.php]: The finder should not be removed

Comment (dereuromark) [inline src/ORM/Behavior/TreeBehavior.php]: Is it? ;)

Comment (lorenzo) [inline src/ORM/Behavior/TreeBehavior.php]: is it what?

Comment (dereuromark) [inline src/ORM/Behavior/TreeBehavior.php]: Are you positive it is removed?

Comment (lorenzo) [inline src/ORM/Behavior/TreeBehavior.php]: Try to find it here https://raw.githubusercontent.com/cakephp/cakephp/master-tree/src/ORM/Behavior/TreeBehavior.php

Comment (dereuromark) [inline src/ORM/Behavior/TreeBehavior.php]: I was surprised that it disappeared. It wasn't me, though.
Blame @markstory for it with https://github.com/cakephp/cakephp/commit/be71d48430f6dec33cc43c36adf2b04dae5ec4f3 :)

Comment (lorenzo) [inline src/ORM/Behavior/TreeBehavior.php]: heh

Comment (markstory) [inline src/ORM/Behavior/TreeBehavior.php]: Derp. I'll restore it.

Comment (markstory) [inline src/ORM/Behavior/TreeBehavior.php]: Fixed now.

**Pull request merged**